src/Stratis.Bitcoin.Features.PoA/PoAMiner.cs

```diff
@@ -75,42 +92,68 @@ private async Task CreateBlocksAsync()
             {
                 try
                 {
-                    // Don't mine in IBD.
-                    if (this.ibdState.IsInitialBlockDownload() && !MineDuringIBD)
+                    // Don't mine in IBD in case we are connected to any node.
+                    if (this.ibdState.IsInitialBlockDownload() && this.connectionManager.ConnectedPeers.Any())
                     {
                         int attemptDelayMs = 20_000;
                         await Task.Delay(attemptDelayMs, this.cancellation.Token).ConfigureAwait(false);
 
                         continue;
                     }
 
+                    uint timeNow = (uint) this.dateTimeProvider.GetAdjustedTimeAsUnixTimestamp();
+                    uint myTimestamp = this.slotsManager.GetMiningTimestamp(timeNow);
+
+                    if (myTimestamp < timeNow)
```

Comment:
Can we do this in `GetMiningTimestamp`? I'd rather make sure we don't have a wacky method in there. :-)

Reply:
It's already done there. Here I have a 2nd layer of protection. That's why the comment says it's a sanity check- that should never happen.

Reply:
I don't see the point of adding it there if I'm perfectly honest.
It's not only that it should never happen, it's already checked. That should be enough.

Reply:
Ok, removed